scripts/src/admin/services.ts

```diff
@@ -428,22 +428,16 @@ export async function getOrders(params: any, query: Paging & { status?: OpenOrde
 }
 
 export async function retryOrder(params: { order_id: string }, query: any): Promise<string> {
-	const order: Order | undefined = await Order.getOne(params.order_id);
+	const order = await Order.getOne(params.order_id);
 
 	if (!order) {
 		throw new Error("order not found: " + params.order_id);
 	}
-	if (order.status !== "failed" || order.type !== "earn") {
+	if (order.status !== "failed" || order.isP2P() || order.contexts[0].type !== "earn") {
 		throw new Error("cant retry non earn or non failed orders");
 	}
 
-	// TODO: what is this for?
-	/*const user = await User.findOneById(order.userId);
-	if (!user) {
-		throw new Error("user not found: " + order.userId);
-	}*/
-
-	await payment.payTo(order.blockchainData.recipient_address!, order.sender!.appId, order.amount, order.id, getDefaultLogger());
+	await payment.payTo(order.blockchainData.recipient_address!, order.contexts[0].user.appId, order.amount, order.id, getDefaultLogger());
```

Comment:
why not contextFor?

Reply:
because you need the user id for that, which we don't have here